plasmapy/physics/parameters.py

```diff
@@ -394,23 +394,113 @@ def thermal_speed(T, particle="e", method="most_probable"):
-                         .format(method)))
+        raise ValueError("Method {method} not supported in thermal_speed")
 
     return V
 
 
+@check_relativistic
+@check_quantity({
+    'T': {'units': units.K, 'can_be_negative': False}
+})
+def kappa_thermal_speed(T, kappa, particle="e", method="most_probable"):
+    r"""Returns the most probable speed for a particle within a Kappa
+    distribution.
+
+    Parameters
+    ----------
+    T : Quantity
+        The particle temperature in either kelvin or energy per particle
+
+    kappa: Quantity
+        The kappa parameter is a dimensionless number which sets the slope
+        of the energy spectrum of suprathermal particles forming the tail
+        of the Kappa velocity distribution function. Kappa must be greater
+        than 3/2.
+
+    particle : string, optional
+        Representation of the particle species (e.g., 'p' for protons, 'D+'
+        for deuterium, or 'He-4 +1' for singly ionized helium-4),
+        which defaults to electrons.  If no charge state information is
+        provided, then the particles are assumed to be singly charged.
+
+    method : string, optional
+        Method to be used for calculating the thermal speed. Options are
+        'most_probable' (default), 'rms', and 'mean_magnitude'. 
+
+    Returns
+    -------
+    V : Quantity
+        particle thermal speed
+
+    Raises
+    ------
+    TypeError
+        The particle temperature is not a Quantity
+
+    UnitConversionError
+        If the particle temperature is not in units of temperature or
+        energy per particle
+
+    ValueError
+        The particle temperature is invalid or particle cannot be used to
+        identify an isotope or particle
+
+    UserWarning
+        If the particle thermal speed exceeds 10% of the speed of light, or
+        if units are not provided and SI units are assumed.
+
+    Notes
+    -----
+    The particle thermal speed is given by:
+
+    .. math::
+        V_{th,i} = \sqrt{(2 \kappa - 3)\frac{2 k_B T_i}{\kappa m_i}}
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> kappa_thermal_speed(5*u.eV, 4, 'p')
+    <Quantity 24467.878463594967 m / s>
+    """
+    # Checking thermal units
+    T = T.to(units.K, equivalencies=units.temperature_energy())
+    if kappa <= 3 / 2:
+        raise ValueError(f"Must have kappa > 3/2, instead of {kappa}, for "
+                         "kappa distribution function to be valid.")
+    # obtaining particle mass
+    try:
+        m = ion_mass(particle)
+    except Exception:
+        raise ValueError(f"Unable to find {particle} mass in thermal_speed")
+    # thermal velocity of Kappa distribution function is just Maxwellian
+    # thermal speed modulated by the following factor.
+    # This is true for the "most probable" velocity, though it may change
+    # for the other two methods. Must be checked!
+    coeff = np.sqrt((kappa - 3 / 2) / kappa)
```

Comment:
Note to self to remember to check this!

Reply:
yes, please 😅

Reply:
So, I'm not sure how to check this. I've tried analytically integrating the 1D distribution function multiplied by v^2 to get the RMS speed, but I get an odd answer. Seems to result in RMS speed = sqrt(k_B * T / m), which is precisely the RMS speed for a 1D Maxwell-Boltzmann distribution.

Then again, this might make sense since the most probable thermal velocity of the kappa distribution is in the distribution itself and ends up canceling all the kappas from the integration.

This is messing with my intuition as I would think that the high energy tails would shift the mean by some factor of kappa to high velocities compared to the Maxwellian distribution, and that it would only converge to the Maxwellian RMS speed for large values of kappa.

Reply:
There seems to be a similar sort of cancellation in 3D as well (as far as the constants and thermal velocities are concerned) though I haven't managed to do the triple integral as it is quite complex and not easily separable into x,y,z components like the Maxwellian.

Reply:
@namurphy thoughts on how to check RMS and mean magnitude speeds for the kappa distribution function?

Perhaps this can be numerically checked as we have 2 hypotheses as to what the speed should be? (i.e., how it is currently defined in kappa thermal speed vs. the analytical approach which seems to yield the Maxwellian thermal speeds for RMS and mean magnitude)

Reply:
Similar test for mean magnitude
```
Numerically obtained vmean: 14850.474638693995
Functionally obtained vmean: 12346.067837955574 m / s
Maxwellian vmean: 15616.677805956093 m / s
```

It's clear that the way the kappa_thermal_speed is currently defined is wrong. The Maxwellian mean magnitude and the integration method are fairly close. I think the integral is off because there is a lot of weight in the tails of a kappa distribution and those parts become hard to integrate due to numerical errors.

So yea looks like the RMS and mean speeds for kappa distribution are equivalent to that of a Maxwellian.

side note: found an error in how mean magnitude velocity was defined in thermal_speed(), so I'll commit the correction.